### PR TITLE
Fixes scream and fart spam (and dead bodies screaming and farting)

### PR DESCRIPTION
**Pull request by tortellinitony**
Fixes issues #10 and #17

### CODE REVIEW

File: code/modules/mob/dead/observer/say.dm [hunk 1]
```diff
@@ -4,6 +4,9 @@
 	if (!message)
 		return
 
+	if(message == "*scream" || message == "*fart")
```
Comment (Alphora) [inline code/modules/mob/dead/observer/say.dm]: Is this even necessary? Should be fixed in the emote.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -151,6 +151,8 @@
 					src << "<span class='notice'>Unusable emote '[act]'. Say *help for a list.</span>"
 
 		if ("fart")
+			if(src.stat != CONSCIOUS)
```
Comment (Alphora) [inline code/modules/mob/living/carbon/human/emote.dm]: Move this to the top of the proc. Emotes shouldn't be doable when unconscious anyway eh?

File: code/modules/mob/dead/observer/say.dm (hunk 1 above)
Comment (tortellinitony) [inline code/modules/mob/dead/observer/say.dm]: Yes, you can spam deadchat really quickly if not

File: code/modules/mob/living/emote.dm [hunk 3]
```diff
@@ -217,6 +217,8 @@
 			m_type = 2
 
 		if ("scream","screams")
+			if(src.stat != CONSCIOUS)
```
Comment (Alphora) [inline code/modules/mob/living/emote.dm]: Move this check to the top of the proc along with a check for dead. See the top of the same proc in code/modules/mob/living/carbon/human/emote.dm

File: code/modules/mob/living/carbon/human/emote.dm (hunk 2 above)
Comment (tortellinitony) [inline code/modules/mob/living/carbon/human/emote.dm]: That might cause issues with other stuff

Comment (Alphora) [inline code/modules/mob/living/carbon/human/emote.dm]: The check is already there for dead. I've checked, it won't cause issues.

Comment (tortellinitony) [inline code/modules/mob/living/carbon/human/emote.dm]: Ok

File: code/modules/mob/dead/observer/say.dm (hunk 1 above)
Comment (Alphora) [inline code/modules/mob/dead/observer/say.dm]: I don't mean the PR, I mean the check for the message is redundant. Should be a check in the actual emote proc.

Comment (tortellinitony) [inline code/modules/mob/dead/observer/say.dm]: The hotkey makes you say *fart, so if it doesn't cause an emote it puts it in the deadchat and you can spam it

File: code/modules/mob/living/carbon/emote.dm [hunk 4]
```diff
@@ -7,6 +7,8 @@
 	var/delay = 5
 	if(spam_flag == 1)
 		return
+	if(stat != CONSCIOUS && (act != "deathgasp") || (status_flags & FAKEDEATH)) //if we're faking, don't emote at all
```
Comment (Alphora) [inline code/modules/mob/living/carbon/emote.dm]: Close but not quite. Make it:
`if(stat == DEAD && (act != "deathgasp") || (status_flags & FAKEDEATH) || stat != CONSCIOUS)`

File: code/modules/mob/living/emote.dm [hunk 5]
```diff
@@ -7,6 +7,8 @@
 /mob/living/emote(act, m_type=1, message = null)
 	if(stat)
 		return
+	if(stat != CONSCIOUS && (act != "deathgasp") || (status_flags & FAKEDEATH)) //if we're faking, don't emote at all
```
Comment (Alphora) [inline code/modules/mob/living/emote.dm]: `if(stat == DEAD && (act != "deathgasp") || (status_flags & FAKEDEATH) || stat != CONSCIOUS)`

File: code/modules/mob/dead/observer/say.dm (hunk 1 above)
Comment (Alphora) [inline code/modules/mob/dead/observer/say.dm]: It should be made to still clip it like it's an emote but not do the action. Other emotes don't show in deadchat and don't need a specific conditional, why would these two?

Comment (Alphora) [inline code/modules/mob/dead/observer/say.dm]: Yeah, it should already remove it. Take this check out and I'll approve.

**Comment by tortellinitony** [comment]
done

**Comment by Alphora** [comment]
Not to be anal but it's still in code/modules/mob/living/say.dm

Get that one real quick please lol

**Comment by tortellinitony** [comment]
I will show you what I mean, let me get a picture

**Comment by tortellinitony** [comment]
The checks need to be there, otherwise it's a spam button

https://vgy.me/VurML7.png

**Comment by Alphora** [comment]
I see now. I'll do some edits to the deadchat myself later to make it a bit more proper. I'll go ahead and merge this in the time being.

**Pull request merged**